### PR TITLE
-Work with databases-

### PR DESCRIPTION
**Pull request by emerix1993**

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -0,0 +1 @@
+BOT_TOKEN = "6434326001:AAF3LFNeBCXRewwGEyaTY5aEp8FAtVUd_Gg"
```
Comment (Vir2S) [inline config.py:1]: This must be in .env file I think

File: main.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from telebot import types
 from telebot.types import InlineKeyboardMarkup, InlineKeyboardButton, ReplyKeyboardMarkup, KeyboardButton, \
     ReplyKeyboardRemove
-from models import User, Base, engine
+from models import User, Base, engine, session
```
Comment (Vir2S) [inline main.py]: Base no need to import, only User.

File: main.py [hunk 3]
```diff
@@ -22,9 +22,11 @@
 procedures = ['Стрижка', 'Гоління', 'Комплекс', 'Написати адміністратору']
```
Comment (Vir2S) [inline main.py]: procedures must be in constants or in config

File: main.py [hunk 4]
```diff
@@ -22,9 +22,11 @@
...
 
 
+phone_number = None  #Global variable to store the phone number
+
 @bot.message_handler(commands=['start'])
 def phone(message):
```
Comment (Vir2S) [inline main.py]: I think naming is wrong. I woudl like to see something like get_phone_number not phone.

File: main.py [hunk 5]
```diff
@@ -22,9 +22,11 @@
...
-    global chat_id
+    global phone_number
```
Comment (Vir2S) [inline main.py]: Bad practice to use global operator.

File: main.py [hunk 6]
```diff
@@ -22,9 +22,11 @@
...
     chat_id = message.chat.id
     keyboard = types.ReplyKeyboardMarkup(row_width=1, resize_keyboard=True)
     button_phone = types.KeyboardButton(text="Відправити номер телефону", request_contact=True)
```
Comment (Vir2S) [inline main.py]: where is return phone_number?

File: main.py [hunk 7]
```diff
@@ -22,9 +22,11 @@
 procedures = ['Стрижка', 'Гоління', 'Комплекс', 'Написати адміністратору']
 
 
+phone_number = None  #Global variable to store the phone number
```
Comment (Vir2S) [inline main.py]: What is it?))

File: main.py [hunk 8]
```diff
@@ -35,6 +37,9 @@ def phone(message):
 @bot.message_handler(content_types=['contact'])
 def contact(message):
```
Comment (Vir2S) [inline main.py]: Do not use naming as variable, functions must have verbs in their names.

File: main.py [hunk 9]
```diff
@@ -35,6 +37,9 @@ def phone(message):
...
     if message.contact is not None:
+        global phone_number
```
Comment (Vir2S) [inline main.py:55]: Bad practice to use global if you can to not to do it.

File: main.py [hunk 10]
```diff
@@ -46,14 +51,23 @@ def contact(message):
 def process_procedure(callback_query, process_date_time=str):
     procedure = callback_query.data
     bot.send_message(callback_query.from_user.id, f"Ви обрали процедуру: {procedure}")
-    new_user = User(chat_id=callback_query.message.chat.id, procedure=procedure, first_name=callback_query.from_user.first_name, last_name=callback_query.from_user.last_name, phone=callback_query.from_user.contact.phone_number)
-    session.add(new_user)
-    session.commit()
-    bot.send_message(callback_query.from_user.id,
-                     "Введіть дату та час візиту у форматі ДД-ММ та час(приклад -> 19.05 15:00)")
-    bot.register_next_step_handler(callback_query.message, process_date_time)
+    global phone_number
```
Comment (Vir2S) [inline main.py]: Bad practice)

File: main.py [hunk 11]
```diff
@@ -46,14 +51,23 @@ def contact(message):
...
+            chat_id=callback_query.from_user.id,
+            first_name=callback_query.from_user.first_name,
+            last_name=callback_query.from_user.last_name,
+            phone=phone_number,
+            procedure=procedure
+        )
+        session.add(new_user)
+        session.commit()
+        bot.send_message(callback_query.from_user.id,
+                         "Введіть дату та час візиту у форматі ДД-ММ та час(приклад -> 19.05 15:00)")
```
Comment (Vir2S) [inline main.py]: In  my opinion better to split all texts into separate file as constants. But not critical.

File: models.py [hunk 12]
```diff
@@ -9,6 +9,8 @@
 
 engine = create_engine('mysql://admin:admin@localhost/barbershop_telegram')
```
Comment (Vir2S) [inline models.py:10]: credentials must be in .env file

File: models.py [hunk 13]
```diff
@@ -24,8 +26,6 @@ class User(Base):
 
     Base.metadata.create_all(engine)
 
-    Session = sessionmaker(bind=engine)
-    session = Session()
 
     def __repr__(self):
```
Comment (Vir2S) [inline models.py:30]: Why __repr__ instead __str__?

File: config.py (hunk 1 above)
Comment (emerix1993) [reply config.py:1]: Good idea))
I don't think about this)

File: main.py (hunk 2 above)
Comment (emerix1993) [reply main.py]: Complete

File: main.py (hunk 5 above)
Comment (emerix1993) [reply main.py]: Maybe, you know alternative way?

File: main.py (hunk 11 above)
Comment (emerix1993) [reply main.py]: This part isn't working, i write this for future changes

File: models.py (hunk 12 above)
Comment (emerix1993) [reply models.py:10]: +

File: models.py (hunk 13 above)
Comment (emerix1993) [reply models.py:30]: Don't understand what you mean)

Comment (Vir2S) [reply models.py:30]: What's the difference between repr and str?

File: main.py (hunk 5 above)
Comment (Vir2S) [reply main.py]: Yes, I know

File: main.py (hunk 11 above)
Comment (Vir2S) [reply main.py]: okay

**Pull request merged**